lib/_emerge/actions.py

```diff
@@ -2655,18 +2655,37 @@ def nice(settings):
 		out = portage.output.EOutput()
 		out.eerror(f"Failed to change nice value to {nice_value}")
 		out.eerror("%s\n" % str(e))
+		return
 
-	autogroup_file = Path("/proc/self/autogroup")
-	try:
-		f = autogroup_file.open("r+")
-	except EnvironmentError:
+	sched_autogroup = Path("/proc/sys/kernel/sched_autogroup_enabled")
+	if not sched_autogroup.is_file():
 		# Autogroup scheduling is not enabled on this system.
 		return
```

Comment:
The kind of TOCTOU race that I want it to handle would be where the `/proc/sys/kernel/sched_autogroup_enabled` state changed right during the critical section here.